scraper.go

```diff
@@ -131,6 +135,59 @@ func scrapeAll(ldapAddr, ldapUser, ldapPass string) error {
 	return errs
 }
 
+func scrapeDial(ldapAddr string, insecure bool) (*ldap.Conn, error) {
+	re := regexp.MustCompile(`^(?:(?P<ldapScheme>ldapi|ldap|ldaps):\/\/)?(?P<ldapHost>(?P<ipv4>(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])|(?P<ipv6>\[[a-z0-9\-._~%!$&'()*+,;=:]+\])|(?P<fqdn>[a-zA-Z0-9\-._~%]+))(?::(?P<ldapPort>\d+))?\/?$`)
+	
+	// match[1]: scheme
+	// match[2]: escaped path, when scheme is ldapi
+	//           hostname, when schemes are ldap and ldaps
+	// match[6]: port number (of type string)
+	if match := re.FindStringSubmatch(ldapAddr); match != nil {
```

Comment:
Why not use the [url.Parse](https://pkg.go.dev/net/url#Parse) ?

Reply:
Basically because I'm new to Go Lang... 😉

Looking at the url.Parse() and URL type, it should be a perfect replacement for the regular expression.